x-pack/plugins/observability_solution/observability_ai_assistant_app/public/hooks/use_local_storage.tsx

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useState, useEffect, useMemo } from 'react';
+
+export function useLocalStorage<T>(key: string, defaultValue: T) {
```

Comment:
I have a vague recollection that we already discussed but I cannot remember the answer: can we use the existing `useLocalStorage` hook? 

```
import useLocalStorage from 'react-use/lib/useLocalStorage';
```

Reply:
Ahh.. is it because the hook from `react-use` doesn't listen to storage events, so it does not work when multiple tabs are open?

Reply:
@dgieselaar has requested that we move away from that hook as that project has many issues.